pettingzoo/mpe/scenarios/group_spread.py

```diff
@@ -1,91 +1,155 @@
+from enum import Enum
+
 import numpy as np
+
 from .._mpe_utils.core import World, Agent, Landmark
 from .._mpe_utils.scenario import BaseScenario
-import random
+
+
+# Predefined colors for agents and landmarks
+COLORS = [
+    np.array([0.25, 0.25, 0.25]),
+    np.array([0.85, 0.25, 0.25]),
+    np.array([0.25, 0.85, 0.25]),
+    np.array([0.25, 0.25, 0.85]),
+    np.array([0.85, 0.85, 0.25]),
+    np.array([0.85, 0.25, 0.85]),
+    np.array([0.25, 0.85, 0.85]),
+    np.array([1.00, 0.00, 0.00]),
+    np.array([0.00, 1.00, 0.00]),
+    np.array([0.00, 0.00, 1.00]),
+]
+
+class ColorConfig(Enum):
+    CONTINUOUS_FIXED = 0    # used fixed set of colors (matched by `num_colors`) which agents observe as continuous vectors
+    ONEHOT_FIXED = 1        # used fixed set of colors (matched by `num_colors`) which agents observe as one-hot vectors
+    CONTINUOUS_RANDOM = 2   # used random colors which agents observe as continuous vectors (not using `num_colors`!)
+
+    @staticmethod
+    def from_str(label):
+        """
+        Returns the ColorConfig enum value corresponding to the given string label.
+        :param label: string label
+        :return: ColorConfig enum value
+        """
+        if label.lower().replace(" ", "_") in ("continuousfixed", "continuous_fixed"):
+            return ColorConfig.CONTINUOUS_FIXED
+        elif label.lower().replace(" ", "_") in ("onehotfixed", "onehot_fixed", "fixed"):
+            return ColorConfig.ONEHOT_FIXED
+        elif label.lower().replace(" ", "_") in ("continuousrandom", "continuous_random", "random"):
+            return ColorConfig.CONTINUOUS_RANDOM
+        else:
+            raise NotImplementedError(f"Unknown color config label '{label}'")
+
+    def get_color_values(self, num_colors):
+        """
+        Generate color values for the given color config and number of colors.
+        :param color_config: ColorConfig enum value
+        :param num_colors: number of colors
+        :return: list of color values (each color value is a numpy array of shape (3,))
+        """
+        assert num_colors is not None, "Number of colors must be specified!"
+        assert num_colors > 0 and num_colors <= len(COLORS), f"Number of colors must be in range [1, {len(COLORS)}]!"
+
+        if self == ColorConfig.CONTINUOUS_RANDOM:
+            # generate random set of colors
+            return [np.random.random(3) for _ in range(num_colors)]
+        elif self == ColorConfig.ONEHOT_FIXED or self == ColorConfig.CONTINUOUS_FIXED:
+            # generate fixed set of colors using only individual color channels 
+            return COLORS[:num_colors]
```

Comment:
Use just first colors from the pre-defined fixed set of colors whenever using standard fixed colors.